### PR TITLE
Fix constructive_solid_geometry.py for non_utf8_error

### PR DESCRIPTION
**Pull request by jj-umn**
python3 read ahead can throw UnicodeDecodeError when reading a Vtk BINARY file.  Catch  UnicodeDecodeError  and read as 'rb'.

(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by jj-umn** [comment]
@gregvonkuster   I'm working on tools for Neper https://github.com/rquey/neper which can output both ASCII and BINARY Vtk files.   I'll try to add datatype tests to this, once I can get the current automated checks to work.

**Comment by gregvonkuster** [comment]
Thanks for this @jj-umn - it looks great.  I wasn't aware that anyone was still using these datatypes. :)

**Comment by gregvonkuster** [comment]
> 



> @gregvonkuster I was thinking of adding Neper datatypes for file formats tesr, tess, and Gmsh msh to constructive_solid_geometry.py if that seems like an appropriate place. https://neper.info/docs/neper.pdf

Sounds fantastic.  In case you aren't aware, this is the viewer for the vtk datatypes - https://github.com/galaxyproject/galaxy/tree/dev/config/plugins/visualizations/csg.  I haven't used it for a while, so I hope it's still functional.

**Comment by jj-umn** [comment]
@gregvonkuster   Is there anyone else that you think should review this?   I'll add neper datatypes in a separate pull request after this is approved.

**Comment by gregvonkuster** [comment]
> @gregvonkuster Is there anyone else that you think should review this? I'll add neper datatypes in a separate pull request after this is approved.

No one that I know of.

**Comment by jj-umn** [comment]
@gregvonkuster  Let me know if there any any changes needed.  Thanks.

File: lib/galaxy/datatypes/constructive_solid_geometry.py
Comment (mvdbeek) [inline lib/galaxy/datatypes/constructive_solid_geometry.py]: There's a few odd or wrong patterns in this PR which I think we can improve by adding a [TextIOWrapper](https://docs.python.org/3/library/io.html#io.TextIOWrapper) to the StringIO class and using that with the `errors='ignore'`. Because I think that's essentially the behavior you're going for here, that is dealing with mixed utf-8 and non-utf-8 binary content, where the sniffer just looks at the utf-8 content. I'll push a commit and you can let me know if that is still correct.

Comment (jj-umn) [reply lib/galaxy/datatypes/constructive_solid_geometry.py]: @mvdbeek  Thanks.   This does require reading the header portions of files that may contain non-header data in either binary or ascii format.   

The sniff_prefix function encountered error from lib/galaxy/datatypes/sniff.py in **class FilePrefix**:  
Is there a way to deal with that more generally in the **class FilePrefix**

> utf-8' codec can't decode byte 0x85 in position 478: invalid start byte

The set_meta functions had the same error when opened as 'r'.  

I'll take a look at TextIOWrapper

Comment (mvdbeek) [reply lib/galaxy/datatypes/constructive_solid_geometry.py]: https://github.com/galaxyproject/galaxy/compare/dev...mvdbeek:constructive_solid_geometry_non_utf8_error?expand=1 is how I would do it. This should tick all your boxes. I did lose your commit authorship unfortunately, but if that looks good to you I would push it here.

Comment (jj-umn) [reply lib/galaxy/datatypes/constructive_solid_geometry.py]: @mvdbeek No worries. I don't care about authorship, just want things to work as best possible.   Can you go ahead and push it?  Many Thanks. JJ

**Review by mvdbeek — APPROVED** [review]
At this point I think we need another committer to review this.

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @jj-umn @gregvonkuster @mvdbeek !

**Pull request merged**
